frappe/model/document.py

```diff
@@ -750,8 +750,10 @@ def check_docstatus_transition(self, docstatus):
 			elif self.docstatus==1:
 				self._action = "submit"
 				self.check_permission("submit")
-			else:
+			elif self.docstatus==2:
 				raise frappe.DocstatusTransitionError(_("Cannot change docstatus from 0 to 2"))
```

Comment:
This can also be improved. Numeric docstatus aren't easily understood by users. We can probably use something like "can't change document status from draft to submitted" 🤔

Reply:
That makes sense to me (though it's draft->cancelled not draft->submitted). Since some people will see this message when using the numbers (during a rest api call, for example), would it make sense to do both?

"Cannot change docstatus from 0 (draft) to 2 (cancelled)"

and 

"Cannot change docstatus from 1 (submitted) to 0 (draft)"

Or something like that?